src/Diagnostics/windowed_time_average.jl

```diff
@@ -13,47 +14,47 @@ mutable struct WindowedTimeAverage{RT, FT, O, R} <: AbstractDiagnostic
                          time_window :: FT
                        time_interval :: FT
                               stride :: Int
+                        field_slicer :: field_slicer
```

Comment:
This is a good point. It seems that we are moving towards the concept that time-averaging is a "transformation of output". In other words, the purpose of the windowed time average is to intercept output and apply averaging to it prior to saving to disk. As a result, the `WindowedTimeAverage` must have a few fields in common with the output writers.

The particular functionality afforded here is time-averaging of slices. This is nice, because it avoids allocating memory for the full field and then discarding much of the data when saving to disk.

I'm open to other abstractions. An alternative possibility is to create a `TimeAveragedField`, similar to `AveragedField`.

Reply:
Yeah I guess it's an evolving design, it looks good so far. `TimeAveragedField` might be a bit weird since you can't call `compute!(taf)` where `taf` is a `TimeAveragedField` as the model does not store state history.

Reply:
Yeah, that's a good point. Time-averaging only makes sense in the context of a simulation / output, so it's perhaps not something we want to include in our (instantaneous in time) field abstraction / operation / analysis ecosystem.